data/tutorials/language/0lg_09_env.md

```diff
@@ -0,0 +1,613 @@
+
+### Passing Tuples
+
+In OCaml, a *tuple* is a fundamental data structure used to group together a fixed number of values, which can be of different types. Tuples are defined using parentheses, and the elements are separated by commas. Here's the basic syntax to create and work with tuples in OCaml:
+```ocaml
+# let space_cat (s, t) = s ^ " " ^ t;;
+val space_cat : string * string -> string = <fun>
+```
+
+It takes a pair of strings and returns a concatenation of those strings with a space character in between.
+```ocaml
+# space_cat ("hello", "world");;
+- : string = "hello world"
+```
+
+The function `space_cat` takes a single parameter, even though it looks like two arguments have been passed: "hello" and "world;" however, since they're enclosed in parentheses with a comma, they are actually a tuple. Tuples with multiple elements are a single value. 
+
+In many imperative languages, this syntax reads as a function call with two parameters; but in OCaml, this syntax denotes applying the function `space_cat` to the tuple `("hello", "world")`. 
+
+### Currying and Uncurrying
+
+In the two previous sections, two kinds of “multiple parameter” functions have been presented.
+- Functions returning a function, such as `( + )`
+- Functions taking a tuple as a parameter, such as `space_cat`
+
+Interestingly, both kinds of functions provide a way to apply them to several arguments while being functions with a single parameter. From this perspective, it makes sense to say: “All functions have a single argument.”
+
+This goes even further. Any function of the same kind can be translated into an equivalent function of the second kind, and conversely. Using OCaml support for higher-order functions, it is possible to define those transformations as functions.
+
+Here is the translation from the first kind to the second kind:
+```ocaml
+# fun f -> fun (x, y) -> f x y;;
+- : ('a -> 'b -> 'c) -> 'a * 'b -> 'c = <fun>
+```
+
+Here is the reverse translation:
+```ocaml
+# fun f -> fun x y -> f (x, y);;
+- : ('a * 'b -> 'c) -> 'a -> 'b -> 'c = <fun>
+```
+
+These translations are attributed to the 20th-century logician [Haskell Curry](https://en.wikipedia.org/wiki/Haskell_Curry). The second translation is called *currying* and the first is called *uncurrying*.
+
+From a typing perspective, this means that for any types `a`, `b`, and `c`, the following types are equivalent:
+- `a -> (b -> c)` &mdash; curried function type
+- `a * b -> c` &mdash; uncurried function type
+
+We will not dive any deeper into the details here, but this equivalence can be formally defined using _ad-hoc_ mathematics.
+
+It is rarely necessary to use the functions above. However, it is important to understand that changing a function may not need any code refactoring. A function can be given several equivalent forms and changed into another, either using refactoring or using a higher-order function. Since functions are values, currying and uncurrying are operations on those values.
```

Comment:
Perhaps Currying/Uncurrying could be put in a more advanced tutorial on functions later in the process. Since this is the first or second tutorial they'll be seeing after the Getting Started, it might be too much detail and derail the happy path.